### PR TITLE
Practice2 task2

### PR DESCRIPTION
**Pull request by tyroff**
Task 2. (20 points) Reflection API usage
Use the project from the previous module [Java Core I] with class hierarchy or another pet project with class hierarchy.
Rewrite your app with Reflection API:
(5 points) Use reflection to instantiate all objects in client code via constructors (with or without arguments)
(5 points) Use reflection API to fill private fields without setters
(5 points) Use reflection API to change method calls from direct call on reflection call in client code
(5 points) Print out the metadata for all classes in application in human-readable format format (about all fields, methods, modifiers and etc.)

### CODE REVIEW

File: src/practice1/task1/transport/Transport.java [hunk 1]
```diff
@@ -59,7 +67,9 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         Transport transport = (Transport) o;
-        return maxCapacityOfPeople == transport.maxCapacityOfPeople && maxSpeed == transport.maxSpeed &&  Objects.equals(mark, transport.mark) && Objects.equals(model, transport.model) && fuelType == transport.fuelType && Objects.equals(countryOfManufacture, transport.countryOfManufacture);
+        return maxCapacityOfPeople == transport.maxCapacityOfPeople && maxSpeed == transport.maxSpeed
```
Comment (UShalamitski) [inline src/practice1/task1/transport/Transport.java:70]: Add equals and hashcode to InventoryEntity and update same methods here by using equals and hashcode from super class

File: src/practice1/task1/transport/WheeledTransport.java [hunk 2]
```diff
@@ -19,10 +34,24 @@ public double getFuelConsumption() {
...
+        if (this == o) return true;
+        if (!(o instanceof WheeledTransport)) return false;
+        if (!super.equals(o)) return false;
+        WheeledTransport that = (WheeledTransport) o;
+        return numberOfWheels == that.numberOfWheels && Double.compare(that.fuelConsumption, fuelConsumption) == 0;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), numberOfWheels, fuelConsumption);
```
Comment (UShalamitski) [inline src/practice1/task1/transport/WheeledTransport.java:48]: Equals and hashcode shouldn't be based on different fields

File: src/practice1/task1/utils/MetaDataOfClassUtil.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package practice1.task1.utils;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+public class MetaDataOfClassUtil {
```
Comment (UShalamitski) [inline src/practice1/task1/utils/MetaDataOfClassUtil.java:6]: missing javadoc

File: src/practice2/task2/Main.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import practice1.task1.transport.WheeledTransport;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Main {
```
Comment (UShalamitski) [inline src/practice2/task2/Main.java:18]: missing javadoc.

File: src/practice1/task1/transport/WheeledTransport.java (hunk 2 above)
Comment (tyroff) [reply src/practice1/task1/transport/WheeledTransport.java:48]: equals has -> if (!super.equals(o)) return false;

File: src/practice1/task1/transport/Transport.java (hunk 1 above)
Comment (tyroff) [reply src/practice1/task1/transport/Transport.java:70]: add

File: src/practice1/task1/utils/MetaDataOfClassUtil.java (hunk 3 above)
Comment (tyroff) [reply src/practice1/task1/utils/MetaDataOfClassUtil.java:6]: add

File: src/practice2/task2/Main.java (hunk 4 above)
Comment (tyroff) [reply src/practice2/task2/Main.java:18]: add

File: src/practice1/task1/transport/WheeledTransport.java (hunk 2 above)
Comment (tyroff) [reply src/practice1/task1/transport/WheeledTransport.java:48]: equals has --> if (!super.equals(o)) return false;

**Pull request merged**